File: findPairSum.py
l=list(map(int,input("enter array").split(" ")))
x=int(input("enter the pair sum"))
d={}
for i in l:
	d[i]=i
ans=[]
for i in d:
	n=x-i
	if(d.get(n)!=None):
		if((i,n) not in ans and (n,i) not in ans):
			ans.append((i,n))
print(ans)

File: FindPairSum.java
import java.util.*;
class FindPairSum{
	public static void main(String args[]){
		Hashtable<Integer,Integer> ht=new Hashtable<Integer,Integer>();
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt();
		for(int i=0;i<n;i++){
			int a=sc.nextInt();
			ht.put(a,a);
		}
		int sum=sc.nextInt();
		Set<Integer> set=ht.keySet();
		Hashtable<Integer,Integer> ans=new Hashtable<Integer,Integer>();
		

		for(Integer key:set){
			if(ht.containsKey(sum-key)){
				if(!ans.containsKey(key) && !ans.containsKey(sum-key)){
					ans.put(key,sum-key);
				}
			}
		}
			System.out.println(ans);
	
}
}